metrics/core.py

```diff
@@ -99,26 +101,38 @@ def search(
         )
         return search_result
 
+
+@dataclass
+class BestChoiceImagesDataset:
+    similars: list[Image.Image]
+    results: list[ScoredPoint]
+
+    @classmethod
     def get_best_choice_for_uploaded_image(
-        self,
-        base_img: Image.Image,
+        cls,
+        client: MetricClient,
+        anchor: Image.Image,
         collection_name: MetricCollections,
         benchmark: int,
         k: int = 25,
-    ) -> tuple[Image.Image, list[Image.Image], list[ScoredPoint]]:
+    ) -> BestChoiceImagesDataset:
         """
         Search for similar images of random image from given collection.
         Returns tuple of images [anchor_image, grid image of k most similar images (the biggest cosine similarity)]
         """
-        results = self.search(base_img, collection_name, limit=k)
-        results_bench = [r for r in results if round(r.score, 4) >= benchmark / 100]
-        if len(results_bench) > 0:
+        results_all = client.search(anchor, collection_name, limit=k)
+        results = [r for r in results_all if round(r.score, 4) >= benchmark / 100]
+        if len(results) > 0:
             imgs = [
                 RESIZE_TRANSFORM(img)
-                for img in env_handler.get_best_score_imgs(results=results_bench)
+                for img in env_handler.get_best_score_imgs(results=results)
             ]
             to_image = torchvision.transforms.ToPILImage()
-            imgs_transformed = [to_image(img) for img in imgs]
+            similars = [to_image(img) for img in imgs]
         else:
-            imgs_transformed = None
-        return base_img, imgs_transformed, results_bench
+            similars = None
```

Comment:
similars = None
if len(results) > 0:
    similars = [to_image(img) for img in imgs]